libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -477,6 +472,18 @@ size_t WiFiClient::write(Stream &stream)
 
 int WiFiClient::read(uint8_t *buf, size_t size)
 {
+    if(_lastReadTimeout != _timeout){
+        if(fd() >= 0){
+            struct timeval timeout_tv;
+            timeout_tv.tv_sec = _timeout / 1000;
+            timeout_tv.tv_usec = (_timeout % 1000) * 1000;
+            if(setSocketOption(SO_RCVTIMEO, (char *)&timeout_tv, sizeof(struct timeval)) >= 0)
```

Comment:
Cosmethic change, but I suggest to use sizeof(timeout_tv)

Reply:
I disagree, as it is good practice to actually use the struct/class for the `sizeof()` instead of the object.
This way you are less likely to end up with the value `4` as result of your `sizeof()` call when working with pointers instead of reference.
Thus this will reduce the chance of programming errors.

Reply:
Imagine you change the type of `timeout_tv` somewhere earlier and forget to replace all `sizeof (struct timeval)` later in the code. Maybe a better thing is to declare const variable/define/constexpr timeval_size just after the `timeout_tv` declaration and then use this variable later in the code.

Reply:
Sure, or maybe there is a compiler flag which can warn when `sizeof()` is given a pointer as that's the actual problem.